File: 002/Problem2.rs
fn main() {
    let start = std::time::SystemTime::now();
    for _i in 0..1000 {
        problem();
    }
    match start.elapsed() {
        Ok(elapsed) => {
            println!("{}", elapsed.as_nanos())
        }
        Err(e) => {
            println!("Error: {:?}", e)
        }
    }
}

fn problem() {
    let mut a = 1;
    let mut b = 2;
    let mut c = 1;
    let mut sum = 0;
    while c < 4_000_000 {
        c = a + b;

        if c % 2 == 0 {
            sum += c;
        }

        a = b;
        b = c;
    }

    println!("{}", sum);
}

File: 002/Problem2.r
problem <- function() {
    a <- 1
    b <- 2
    sum <- 2
    c <- a + b
    while(c < 4000000) {
        c <- a + b
        if(c %% 2 == 0) {
            sum <- sum + c
        }
        a <- b
        b <- c
    }
    print(sum)
}
num_loops <- 10000
start.time <- Sys.time()
for(i in 1:num_loops) {
    problem()
}
end.time <- Sys.time()
time.taken <- end.time - start.time
print(time.taken / num_loops)